service/grpc/service.go

```diff
@@ -57,7 +62,14 @@ func newService(lis net.Listener, opts ...grpc.ServerOption) *Server {
 		authToken:       os.Getenv(common.AppAPITokenEnvVar),
 	}
 
-	gs := grpc.NewServer(opts...)
+	var gs *grpc.Server
+
+	if grpcServer != nil {
+		gs = grpcServer
+	} else {
+		gs = grpc.NewServer(opts...)
+	}
```

Comment:
Prob don't really need a new variable here.

```suggestion
	if grpcServer == nil {
		grpcServer = grpc.NewServer(opts...)
	}
```